executor/memory_test.go

```diff
@@ -141,4 +141,12 @@ func (s *testMemoryLeak) TestGlobalMemoryTrackerOnCleanUp(c *C) {
 	tk.MustExec("update t set id = 6 where id = 3")
 	afterConsume = executor.GlobalMemoryUsageTracker.BytesConsumed()
 	c.Assert(originConsume, Equals, afterConsume)
+
+	// assert stmt trackers haven't been detaced before Next
+	rs, err := tk.Exec("select * from t t1 join t t2 join t t3")
+	c.Assert(err, Equals, nil)
+	defer rs.Close()
+	req := rs.NewChunk()
```

Comment:
This interface has been modified recently
```suggestion
	req := rs.NewChunk(nil)
```